src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -832,8 +832,20 @@ public async Task<SolutionUpdate> EmitSolutionUpdateAsync(Solution solution, Sol
                     }
 
                     var projectSummary = GetProjectAnalysisSymmary(changedDocumentAnalyses);
-                    if (projectSummary == ProjectAnalysisSummary.CompilationErrors || projectSummary == ProjectAnalysisSummary.RudeEdits)
+                    if (projectSummary == ProjectAnalysisSummary.CompilationErrors)
+                    {
+                        isBlocked = true;
+                    }
+                    else if (projectSummary == ProjectAnalysisSummary.RudeEdits)
                     {
+                        foreach (var analysis in changedDocumentAnalyses)
+                        {
+                            if (analysis.RudeEditErrors.Length > 0)
+                            {
+                                documentsWithRudeEdits.Add((analysis.DocumentId, analysis.RudeEditErrors));
+                            }
+                        }
+
                         isBlocked = true;
                     }
```

Comment:
can you doc the differences here.  i.e. why we track the docs for rude edits, but not the docs for compiler errors.

Reply:
I'll follow up.